core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -52,6 +53,30 @@
   private final boolean mClusterDefaultsLoaded;
 
   /**
+   * WARNING: This API is unstable and may be changed in a future minor release.
+   *
+   * Users should use this API to obtain a configuration for modification before passing to a
+   * FileSystem constructor. The default configuration contains all default configuration params
+   * and configuration properties modified in the alluxio-site.properties file.
+   *
+   * Example usage:
+   *
+   * InstancedConfiguration conf = InstancedConfiguration.defaults();
+   * conf.set(...);
+   * FileSystem fs = FileSystem.Factory.create(conf);
+   *
+   * @return an instanced configuration preset with defaults
+   */
+  public static InstancedConfiguration defaults() {
+    return new InstancedConfiguration(ConfigurationUtils.defaults());
+  }
+
+  /**
+   * WARNING: This API is not intended to be used outside of internal Alluxio code and may be
```

Comment:
How would you feel about adding to the `alluxio.annotation` package? We use `@Deprecated` and `@PublicApi` quite a bit. Maybe this could warrant a new `@Unstable`? Or I think guava also has some similar annotations too.

I don't feel too strongly, but I think using annotations for things like this is pretty standard.

Reply:
I think its a good idea but wouldn't really want to introduce it now since we would need to re-evaluate the codebase and apply it everywhere

Reply:
If you feel it's too much effort at this point then I am okay without it. However annotating the API may give developers' IDEs information about using unstable or deprecated APIs. I know that in the guava library when things are marked as experimental or unstable, I get a warning in my IDE about using it.

Reply:
If we mark some apis as unstable, I think it will make all the other ones implicitly stable.

Reply:
Isn't that standard though?